rust/connlib/snownet/src/allocation.rs

```diff
@@ -393,9 +393,10 @@ impl Allocation {
         }
 
         if let Some(refresh_at) = self.refresh_allocation_at() {
-            if now > refresh_at {
+            if now >= refresh_at {
                 tracing::debug!("Allocation is due for a refresh");
                 self.authenticate_and_queue(make_refresh_request());
+                self.allocation_lifetime = None;
```

Comment:
I don't think this is the correct way to solve this. You are creating a state disparity here with the relay. We still have an allocation and suddenly we don't know its lifetime anymore.

Can w instead reset this any time we also reset the candidates? We should keep the data in here consistent.